Weaver/RealmWeaver.Fody/ImportedReferences.cs

```diff
@@ -0,0 +1,353 @@
+
+        public TypeReference Realm { get; private set; }
+
+        public MethodReference Realm_Add { get; private set; }
+
+        public TypeReference RealmObject { get; private set; }
+
+        public MethodReference RealmObject_get_IsManaged { get; private set; }
+
+        public MethodReference RealmObject_get_Realm { get; private set; }
+
+        public MethodReference RealmObject_RaisePropertyChanged { get; private set; }
+
+        public MethodReference RealmObject_GetObjectValue { get; private set; }
+
+        public MethodReference RealmObject_SetObjectValue { get; private set; }
+
+        public MethodReference RealmObject_GetListValue { get; private set; }
+
+        public MethodReference RealmObject_GetBacklinks { get; private set; }
+
+        public TypeReference IRealmObjectHelper { get; private set; }
+
+        public TypeReference PreserveAttribute { get; private set; }
+
+        public MethodReference PreserveAttribute_Constructor { get; private set; }
+
+        public MethodReference PreserveAttribute_ConstructorWithParams { get; private set; }
+
+        public TypeReference WovenAttribute { get; private set; }
+
+        public MethodReference WovenAttribute_Constructor { get; private set; }
+
+        public TypeReference WovenPropertyAttribute { get; private set; }
+
+        public MethodReference WovenPropertyAttribute_Constructor { get; private set; }
+
+        public TypeReference PropertyChanged_DoNotNotifyAttribute { get; private set; }
+
+        public MethodReference PropertyChanged_DoNotNotifyAttribute_Constructor { get; private set; }
+
+        protected ImportedReferences(ModuleDefinition module)
+        {
+            Module = module;
+
+            IEnumerableOfT = new TypeReference("System.Collections.Generic", "IEnumerable`1", Module, Types.CoreLibrary);
+            IEnumerableOfT.GenericParameters.Add(new GenericParameter(IEnumerableOfT));
+
+            ICollectionOfT = new TypeReference("System.Collections.Generic", "ICollection`1", Module, Types.CoreLibrary);
+            ICollectionOfT.GenericParameters.Add(new GenericParameter(ICollectionOfT));
+
+            IListOfT = new TypeReference("System.Collections.Generic", "IList`1", Module, Types.CoreLibrary);
+            IListOfT.GenericParameters.Add(new GenericParameter(IListOfT));
+
+            System_Type = new TypeReference("System", "Type", Module, Types.CoreLibrary);
+
+            System_DateTimeOffset = new TypeReference("System", "DateTimeOffset", Module, Types.CoreLibrary, true);
+
+            // If the assembly has a reference to PropertyChanged.Fody, let's look up the DoNotNotifyAttribute for use later.
+            var PropertyChanged_Fody = Module.AssemblyReferences.SingleOrDefault(a => a.Name == "PropertyChanged");
+            if (PropertyChanged_Fody != null)
+            {
+                InitializePropertyChanged_Fody(PropertyChanged_Fody);
```

Comment:
This method is fairly small, we could inline it I guess. Alternatively, we could move the check for PropertyChanged_Fody inside the method as well.